syntax/symbol.rs

```diff
@@ -34,18 +29,47 @@ impl Symbol {
         self.0.write_fmt(format_args!("{}", segment)).unwrap();
         assert!(self.0.len() > len_before);
     }
+
+    pub fn from_idents<'a, T: Iterator<Item = &'a Ident>>(it: T) -> Self {
+        let mut symbol = Symbol(String::new());
+        for segment in it {
+            segment.write(&mut symbol);
+        }
+        assert!(!symbol.0.is_empty());
+        symbol
+    }
+
+    /// For example, for taking a symbol and then making a new symbol
+    /// for a vec of that symbol.
+    pub fn prefix_with(&self, prefix: &str) -> Symbol {
+        Symbol(format!("{}{}", prefix, self.to_string()))
+    }
+}
+
+pub trait Segment {
+    fn write(&self, symbol: &mut Symbol);
 }
 
-pub trait Segment: Display {
+impl Segment for str {
+    fn write(&self, symbol: &mut Symbol) {
+        symbol.push(&self);
+    }
+}
+impl Segment for usize {
+    fn write(&self, symbol: &mut Symbol) {
+        symbol.push(&self);
+    }
+}
+impl Segment for Ident {
+    fn write(&self, symbol: &mut Symbol) {
+        symbol.push(&self);
+    }
+}
+impl Segment for Symbol {
```

Comment:
Why not just `impl Display for CppName` instead so the trait can keep the `Display` bound and the default method implementation?

Reply:
That one is intentional - https://github.com/dtolnay/cxx/pull/370/files#diff-eebf4e8e2220ce00f9803b5fe639e70ae3c646f75d621bd50c76bdcc30f867a8R53